src/threshold_sign.rs

```diff
@@ -99,28 +103,42 @@ impl<N: NodeIdT> DistAlgorithm for ThresholdSign<N> {
 }
 
 impl<N: NodeIdT> ThresholdSign<N> {
-    /// Creates a new instance of `ThresholdSign`, with the goal to collaboratively sign `msg`.
-    pub fn new<M: AsRef<[u8]>>(netinfo: Arc<NetworkInfo<N>>, msg: M) -> Self {
+    /// Creates a new instance of `ThresholdSign`, with the goal to collaboratively sign `doc`.
+    pub fn new(netinfo: Arc<NetworkInfo<N>>) -> Self {
         ThresholdSign {
             netinfo,
-            msg_hash: hash_g2(msg),
+            doc_hash: None,
             received_shares: BTreeMap::new(),
             had_input: false,
             terminated: false,
         }
     }
 
+    /// Sets doc_hash. Signature shares can only be sent after this function is completed.
+    pub fn set_document<M: AsRef<[u8]>>(&mut self, doc: M) -> Result<()> {
```

Comment:
Every call to `set_document` follows a call to `ThresholdSign::new`. Can you call it from `new` instead or explain why you chose to keep them separate? It would look better if there were no separate post-initialization step.

Reply:
But in our use of `ThresholdDecryption`, it's different. And this issue is about bringing the two APIs in line with each other. So even if in our code, `new` and `set_document` are always called together, having them separate might be useful for external users.

Reply:
I'm not entirely convinced though that you couldn't have either `new_with_document` or a builder which makes setting the document a requirement.

Reply:
True, we could add a second constructor that sets the doc on creation. To keep the API consistent we would want to do it for both modules. Is that extra function desired?

Reply:
That's true, that would be convenient. I'd add it to both modules, yes.

Reply:
Change implemented, both modules now have a `new_with_{document,ciphertext}` method.

Reply:
Nice one. Thanks!